test-result-summary-client/src/PerfCompare/PerfCompare.jsx

```diff
@@ -429,34 +464,54 @@ export default class PerfCompare extends Component {
                 curRawValues = {
-                    "baseline": this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].value,
-                    "test": this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].value
+                    "baseline": {
+                        "overview": this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].value,
+                        "iteData": (this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].testsData === undefined) ? childBuildBaselineData : this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].testsData[j].value
+                     },
+                    "test": {
+                        "overview": this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].value,
+                        "iteData": (this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[i].testsData === undefined) ? childBuildTestData : this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[i].testsData[j].value
+                     }
                 };
 
                 try {
-                    curBaselineScore = math.round((this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].mean), 3);
+                    curBaselineScore = math.round((this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].value.mean), 3);
                 } catch (roundBaselineScoreError) {
                     curBaselineScore = "error";
                 }
-                
+
                 try {
-                    curTestScore = math.round((this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].mean), 3);
+                    curTestScore = math.round((this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].value.mean), 3);
                 } catch (roundTestScoreError) {
                     curTestScore = "error";
                 }
 
                 try {
-                    curBaselineCI = math.round((this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].ci), 3);
+                    curBaselineCI = math.round((this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].value.CI), 3);
                     curBaselineCI += "%";
                 } catch(e) {
                     curBaselineCI = "Not found";
                 }
 
                 try {
-                    curTestCI = math.round((this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].ci), 3);
+                    curTestCI = math.round((this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].value.CI), 3);
```

Comment:
Can we handle all number decimal format in table Render? There is no need to handle the baseline and test value separately.

Reply:
Since all kinds of numbers (including `mean`, `ci`) need to be packed into`curMetricTable` variable before going to the table render. So I keep it here to have less modification. If you think it is a must to move in, I might need more changes on the already packed frame.